source/HailPanel.cpp

```diff
@@ -265,7 +266,11 @@ bool HailPanel::KeyDown(SDL_Keycode key, Uint16 mod, const Command &command, boo
 			message = "Thank you for granting us our freedom!";
 		}
 		else
-			message = planet->DemandTribute(player);
+			GetUI()->Push(new Dialog(this, &HailPanel::TributeConfirmed,
+				"Demanding tribute may cause this planet to launch defense fleets to fight you. "
+				"Those defense fleets may be from another government than the planet. "
```

Comment:
I don't get the point of that sentence, it sounds overly technical if that makes sense?  Maybe instead add something like that it would cause a major reputation? just my 2c

Reply:
I changed the message and made "cancel" the default button. Also, I changed it so the message only shows up before the tribute battle. Once the tribute battle has begun, the message serves no purpose.

![tribute-mk2](https://github.com/endless-sky/endless-sky/assets/116329264/adda0da6-27a1-4631-825e-f77395673f29)

Reply:
IMO it still doesn't sound great, particularly this sentence: `Once you defeat all of the tribute fleets, demand tribute again to receive a daily payment.`
It sounds like there was a rule (I know there is, but from the player's perspective it's not obvious) that dominated planets always give you _daily payments_. Also the "demand tribute again" thing sounds a bit too technical - if you demand tribute and the only thing they give you is a battle, it's pretty obvious you need to resend your demand, but idk.

Reply:
Do you have suggestions on how to reword it?

The player does need to know to keep making the demand. The timing of your second demand isn't immediately apparent. Of course, I could remove that sentence and let the player figure it out.

Reply:
> Demanding tribute may cause this planet to launch defense fleets to fight you.  After battling the fleets, you can keep demanding tribute until the planet relents.
> This barbaric act may hurt your reputation severely. Do you want to proceed?

Reply:
Could you make the first part a `help` message (activated the first time the player closes the hail panel after "Our defense fleet will make short work of you" response) and leave only the reputation warning?

Reply:
No, because you don't know yet whether the planet will accept the challenge.

Reply:
wdym?

Reply:
This warning happens before the player actually sends the message.

Reply:
In other words, the player has not demanded tribute yet. The warning is to not demand tribute unless you're ready for what is to come. They won't know whether the demand is met until they make the demand.